crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -2473,16 +2473,32 @@ impl<'db> TypeInferenceBuilder<'db> {
         except_handler_definition: &ExceptHandlerDefinitionKind,
         definition: Definition<'db>,
     ) {
+        fn extract_tuple_specialization<'db>(db: &'db dyn Db, ty: Type<'db>) -> Option<Type<'db>> {
+            let class = ty.into_nominal_instance()?.class();
+            if !class.is_known(db, KnownClass::Tuple) {
+                return None;
+            }
+            let ClassType::Generic(class) = class else {
+                return None;
+            };
+            let specialization = class.specialization(db).types(db)[0];
```

Comment:
I guess we can be sure that any generic alias will always have at least arity-1 specialization, and this will never panic?

(Leaving aside the fact that we already know its `KnownClass::Tuple`, which can only violate our assumptions if a custom typeshed were used.

Reply:
Ah, I forgot about custom typesheds -- I was just justifying this to myself on the grounds of "we know how many type parameters `tuple` has". But yes, I think you're right that any generic alias will have at least arity-1 specialisation, so I don't think this can ever panic!